query/options.go

```diff
@@ -20,6 +20,10 @@ type Options struct {
 	DecodingConcurrency      int
 }
 
+func (o *Options) NumShards() int {
+	return max(o.DecodingConcurrency, 1)
+}
```

Comment:
I am thinking this shard size could be different for different operators? I wonder if the default value 10 works for all operators